### ISSUE TITLE
Unable to launch this plugin in Kicad 5.99

### ISSUE THREAD

**Issue by t123yh** [Bouni/kicad-jlcpcb-tools#64]
Version info (built from kicad-nightly aur):
```
Application: KiCad

Version: (5.99.0-12936-g328d3d0d77), release build

Libraries:
	wxWidgets 3.0.5
	libcurl/7.79.1 OpenSSL/1.1.1l zlib/1.2.11 brotli/1.0.9 zstd/1.5.0 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.45.0

Platform: Linux 5.10.68-1-MANJARO x86_64, 64 bit, Little endian, wxGTK, KDE, x11

Build Info:
	Date: Oct 25 2021 02:17:51
	wxWidgets: 3.0.5 (wchar_t,wx containers,compatible with 2.8) GTK+ 3.24
	Boost: 1.76.0
	OCC: 7.5.0
	Curl: 7.79.1
	ngspice: 35
	Compiler: GCC 11.1.0 with C++ ABI 1016

Build settings:
	KICAD_USE_OCC=ON
	KICAD_SPICE=ON
```

When I click the plugin icon, Kicad exits with an xcb error:
```
2021-10-29 14:35:28,258 [INFO] Load corrections from https://raw.githubusercontent.com/matthewlai/JLCKicadTools/master/jlc_kicad_tools/cpl_rotations_db.csv
2021.10.29 14:35:28 - INFO - Load corrections from https://raw.githubusercontent.com/matthewlai/JLCKicadTools/master/jlc_kicad_tools/cpl_rotations_db.csv
2021-10-29 14:35:28,261 [DEBUG] Starting new HTTPS connection (1): raw.githubusercontent.com:443
2021.10.29 14:35:28 - DEBUG - Starting new HTTPS connection (1): raw.githubusercontent.com:443
2021-10-29 14:36:32,898 [DEBUG] https://raw.githubusercontent.com:443 "GET /matthewlai/JLCKicadTools/master/jlc_kicad_tools/cpl_rotations_db.csv HTTP/1.1" 200 565
2021.10.29 14:36:32 - DEBUG - https://raw.githubusercontent.com:443 "GET /matthewlai/JLCKicadTools/master/jlc_kicad_tools/cpl_rotations_db.csv HTTP/1.1" 200 565
2021-10-29 14:36:32,981 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,981 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,981 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,982 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,982 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,982 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,982 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,983 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,983 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,983 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,983 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,984 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,984 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,984 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,984 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,984 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,985 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,985 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,985 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,985 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,986 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,986 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,986 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,986 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,987 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,987 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,987 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,987 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,988 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,988 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,988 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,988 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,988 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,989 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,989 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,989 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,989 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,990 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,990 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,991 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,991 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,991 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,991 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,991 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,992 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,992 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,992 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,992 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,993 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,993 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,993 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,993 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,994 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,994 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,994 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,994 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,994 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,995 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,995 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,995 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,995 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,996 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,996 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,996 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,996 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,997 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,997 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,997 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,997 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,998 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,998 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,998 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,998 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,998 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,999 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,999 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,999 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:32,999 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:32 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,000 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,000 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,001 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,001 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,001 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,001 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,001 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,002 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,002 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,002 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,002 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,003 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,003 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,003 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,003 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,004 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,004 [ERROR] no such table: jlcpcb_parts
2021.10.29 14:36:33 - ERROR - no such table: jlcpcb_parts
2021-10-29 14:36:33,025 [DEBUG] Start download
2021.10.29 14:36:33 - DEBUG - Start download
2021-10-29 14:36:33,028 [DEBUG] Starting new HTTPS connection (1): jlcpcb.com:443
2021.10.29 14:36:33 - DEBUG - Starting new HTTPS connection (1): jlcpcb.com:443
2021-10-29 14:36:34,033 [DEBUG] https://jlcpcb.com:443 "GET /componentSearch/uploadComponentInfo HTTP/1.1" 200 87497702
2021.10.29 14:36:34 - DEBUG - https://jlcpcb.com:443 "GET /componentSearch/uploadComponentInfo HTTP/1.1" 200 87497702
2021-10-29 14:36:34,035 [DEBUG] Download success
2021.10.29 14:36:34 - DEBUG - Download success
[xcb] Unknown sequence number while processing reply
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
kicad: xcb_io.c:725：_XReply: assertion ‘!xcb_xlib_threads_sequence_lost’ failed.
fish: Job 1, 'kicad-nightly' terminated by signal SIGABRT (Abort)
```

I wonder if some multithreading is not handled correctly in the plugin?

**Comment by Bouni** [comment]
You mean an wx event that the main task then sends to the logger?

**Comment by t123yh** [comment]
> You mean an wx event that the main task then sends to the logger?

I don't quite understand what you refer to. From my basic understanding, every UI operation should be done in UI thread. The logger handler2, which will operate the logbox, may be called from the downloader thread, thus violates this rule. I think we should not operate logbox directly in handler2. Rather, we should post a event to the UI thread in the log handler, and add log to the logbox within the UI thread.

**Comment by Bouni** [comment]
Sounds right, I'll have a look into this next week!

**Comment by Bouni** [comment]
This should be solved in [2021.12.1](https://github.com/Bouni/kicad-jlcpcb-tools/tree/2021.12.1)
